src/utils/indexer/substrate/index.ts

```diff
@@ -0,0 +1,121 @@
+import { DepositDataToSave, ProposalExecutionDataToSave } from "../../../indexer/services/substrateIndexer/substrateTypes";
+
+export async function saveProposalExecution(
+  proposalExecutionData: ProposalExecutionDataToSave,
+  executionRepository: ExecutionRepository,
+  transferRepository: TransferRepository,
+): Promise<void> {
+  const { originDomainId, depositNonce, txIdentifier, blockNumber, timestamp
+  } = proposalExecutionData
+
+  let transfer = await transferRepository.findByNonceFromDomainId(
+    Number(depositNonce),
+    originDomainId
+  )
+
+  // there is no transfer but still we have proposal execution
+  if (!transfer) {
+    try {
+      await transferRepository.insertExecutionTransfer({
+        depositNonce: Number(depositNonce),
+        fromDomainId: originDomainId,
+        timestamp,
+        resourceID: null
+      })
+    } catch (e) {
+      logger.error(`Error inserting substrate proposal execution transfer: ${e}`)
+    }
+  } else {
+    try {
+      await transferRepository.updateStatus(TransferStatus.executed, transfer.id)
+    } catch (e) {
+      logger.error(`Error updating substrate proposal execution transfer: ${e}`)
+    }
+  }
+
+  const execution = {
+    id: new ObjectId().toString(),
+    transferId: transfer!.id,
+    type: "fungible", // TODO: remove this hardcoded string
+    txHash: txIdentifier,
+    blockNumber: blockNumber
+  }
+
+  await executionRepository.insertExecution(execution)
+}
+
+export async function saveDeposit(
+  substrateDepositData: DepositDataToSave,
+  transferRepository: TransferRepository,
+  depositRepository: DepositRepository
+): Promise<void> {
+  const { destDomainId: destinationDomainId, depositNonce, txIdentifier, blockNumber, depositData, handlerResponse, sender, resourceId, timestamp
+  } = substrateDepositData
+
+  const decodedAmount = getDecodedAmount(depositData)
+  
+  const transferData = {
+    id: new ObjectId().toString(),
+    depositNonce: Number(depositNonce),
+    sender: sender,
+    amount: decodedAmount,
+    status: TransferStatus.pending,
+    resource: {
+      connect: {
+        id: resourceId
+      }
+    },
+    fromDomain: {
+      connect: {
+        id: '3'
+      },
+    },
+    toDomain: {
+      connect: {
+        id: destinationDomainId
+      }
+    },
+    timestamp: timestamp,
+  }
+
+  let insertedTransfer: Transfer | undefined
+  try {
+    insertedTransfer = await transferRepository.insertSubstrateDepositTransfer(transferData)
+  } catch (e) {
+    logger.error(`Error inserting substrate deposit: ${e}`)
+  }
+
+  if (insertedTransfer !== undefined) {
+    const deposit = {
+      id: new ObjectId().toString(),
+      type: "fungible", // TODO: remove this hardcoded string
+      txHash: txIdentifier,
+      blockNumber: blockNumber,
+      depositData: depositData,
+      handlerResponse: handlerResponse,
+      transferId: insertedTransfer!.id,
+    }
+
+    try {
+      await depositRepository.insertDeposit(deposit)
```

Comment:
I think it is possible that the transfer for this deposit is already inserted if the indexer fetched the proposalExecuted event on the other chain before it fetches the deposit event on this chain. You should probably search for the transfer and insert it only if the transfer does not exist

Reply:
done, I'm searchin the transfer and updating it